src/anchor.typ

```diff
@@ -0,0 +1,101 @@
+#import "@preview/oxifmt:0.2.0": strfmt
+
+#import "util.typ"
+#import "intersection.typ"
+#import "drawable.typ"
+#import "path-util.typ"
+#import "matrix.typ"
+#import "vector.typ"
+
+// #let calculate(func, anchor-names, anchor, transform: none, name: none, default: none) = {
+//   if anchor == () {
+//     return anchor-names
+//   }
+//   if anchor == "default" {
+//     anchor = default
+//   }
+//   assert(
+//     anchor in anchor-names,
+//     message: strfmt("Anchor '{}' not in anchors {}", anchor, repr(anchor-names)) + if name != none { strfmt(" for element '{}'", name) }
+//   )
+
+//   let out = func(anchor)
+//   return if transform != none {
+//     util.apply-transform(
+//       transform,
+//       out
+//     )
+//   } else {
+//     out
+//   }
+// }
+
+#let setup(callback, anchor-names, default: none, transform: none, name: none, offset-anchor: none) = {
```

Comment:
I have to say, I do not really get this function.
How can I do calculation on (transformed) coordinates for anchor creation?
Example: Line midpoint (which got removed, why?).

Reply:
I've added some docs to this but it mainly returns another function that can be called to get the position of an anchor. Is there a situation where you would need to use transformed coordinates for anchor creation? Even so you can transform the anchors before and leave `transfrom` as `none`.

Also line never had a midpoint anchor? do you mean `center`? which it really shouldn't have. `mid` can be added with path anchors though!

Reply:
Ok. Can you give an example how one could implement a custom anchor that gets calculated in the anchor callback? Something like `mid`?